cirq-core/cirq/transformers/eject_phased_paulis.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transformer pass that pushes 180° rotations around axes in the XY plane later in the circuit."""
+
+from typing import Optional, cast, TYPE_CHECKING, Iterable, Tuple, Dict
+import sympy
+
+from cirq import circuits, ops, value, protocols
+from cirq.transformers import transformer_api, transformer_primitives
+from cirq.transformers.analytical_decompositions import single_qubit_decompositions
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@transformer_api.transformer
+def eject_phased_paulis(
+    circuit: 'cirq.AbstractCircuit',
+    *,
+    context: Optional['cirq.TransformerContext'] = None,
+    atol: float = 1e-8,
+    eject_parameterized: bool = False,
+) -> 'cirq.Circuit':
+    """Transformer pass to push X, Y, and PhasedX gates towards the end of the circuit.
```

Comment:
Added support for `PhasedXZ` gates with z_exponent = 0 (which are equivalent to PhasedX gates). Added a test which demonstrates how to deal with a general `PhasedXZGate`, i.e. 
run `cirq.eject_z(cirq.eject_phased_paulis(cirq.eject_z(c)))`. 
1. `cirq.eject_z(c)` : This ejects Z gates and ensures that the `z_component` of `PhasedXZGate` is 0. 
2. `cirq.eject_phased_paulis(c)`: This ejects `PhasedXPowGates`, potentially leaving z phased from CZ phase kickback, which will then be removed by the 3rd `cirq.eject_z()`.

Reply:
Awesome, could we also support the case where axis_exponent=0 and x_exponent=0 but z_exponent can be whatever we want ? This seems like it might be able to be worked into `_try_get_known_z_half_turns` without much difficulty ?